lib/rubygems/sigstore/openid.rb

```diff
@@ -91,31 +91,37 @@ def get_token()
     end
 
     webserv.abort_on_exception = true
+    redirect_uri = "http://localhost:" + server_addr
 
     client = OpenIDConnect::Client.new(
-        authorization_endpoint: oidc_discovery.authorization_endpoint,
-        identifier: config.oidc_client,
-        redirect_uri: "http://localhost:" + server_addr,
-        secret: config.oidc_secret,
-        token_endpoint: oidc_discovery.token_endpoint,
-      )
+      authorization_endpoint: oidc_discovery.authorization_endpoint,
+      identifier: config.oidc_client,
+      redirect_uri: redirect_uri,
+      secret: config.oidc_secret,
+      token_endpoint: oidc_discovery.token_endpoint,
+    )
 
     authorization_uri = client.authorization_uri(
-        scope: ["openid", :email],
-        state: session[:state],
-        nonce: session[:nonce],
-        code_challenge_method: pkce[:method],
-        code_challenge: pkce[:challenge],
-      )
+      scope: ["openid", :email],
+      state: session[:state],
+      nonce: session[:nonce],
+      code_challenge_method: pkce[:method],
+      code_challenge: pkce[:challenge],
+    )
 
     begin
-      Launchy.open(authorization_uri)
-  rescue
-    # NOTE: ignore any exception, as the URL is printed above and may be
-    #       opened manually
-    puts "Cannot open browser automatically, please click on the link below:"
-    puts ""
-    puts authorization_uri
+      if ENV["SIGSTORE_TEST"]
```

Comment:
These two requests mimic the _browser -> sigstore OAuth2 -> OIDC provider -> browser_ dance just enough to fool our redirect request handler.

I think we can get rid of the first one.

Reply:
Launchy does not appear to have a noop or test mode that simply makes a direct request without opening an external app.  Maybe we could temporarily redefine `Launchy.open` for testing?